stacktrace.go

```diff
@@ -199,25 +197,15 @@ func fileContext(filename string, line, context int) ([][]byte, int) {
 	return lines[start:end], idx
 }
 
-var trimPaths []string
-
-// Try to trim the GOROOT or GOPATH prefix off of a filename
+// trimPath tries to trim the GOROOT or GOPATH prefix off of a filename.
+//
+// We don't use `go/build` here as it will look at the _current_ values,
+// so instead we just assume the last `/src/`.
 func trimPath(filename string) string {
-	for _, prefix := range trimPaths {
-		if trimmed := strings.TrimPrefix(filename, prefix); len(trimmed) < len(filename) {
-			return trimmed
-		}
-	}
-	return filename
-}
-
-func init() {
-	// Collect all source directories, and make sure they
-	// end in a trailing "separator"
-	for _, prefix := range build.Default.SrcDirs() {
-		if prefix[len(prefix)-1] != filepath.Separator {
-			prefix += string(filepath.Separator)
-		}
-		trimPaths = append(trimPaths, prefix)
+	const src = "/src/"
```

Comment:
that said, even if `/src/` doesn't work for some platform or cross-compilation combinations, it should still probably do better than `go/build`'s user-specific paths, so does it make sense to call this an incremental improvement even if there's potentially further we could go?